src/miner_keys.erl

```diff
@@ -100,40 +106,30 @@ keys({ecc, Props}) when is_list(Props) ->
                           {ok, Sig} = miner_ecc_worker:sign(Bin),
                           Sig
                   end,
-       onboarding_key => OnboardingKey
+       onboarding_key => libp2p_crypto:pubkey_to_b58(OnboardingKey)
      }.
 
+-spec key_config() -> key_configuration().
+key_config() ->
+    BaseDir = application:get_env(blockchain, base_dir, "data"),
+    case application:get_env(blockchain, key, undefined) of
+        undefined -> {file, BaseDir};
+        KC -> KC
+    end.
 
 %% @doc prints the public hotspot and onboadring key in a file:consult
 %% friendly way to stdout. This is used by other services (like
 %% gateway_config) to get read access to the public keys
 print_keys(_) ->
-    BaseDir = application:get_env(blockchain, base_dir, "data"),
-    KeyConfig = case application:get_env(blockchain, key, undefined) of
-                    undefined -> {file, BaseDir};
-                    KC -> KC
-                end,
     #{
        pubkey := PubKey,
        onboarding_key := OnboardingKey
-     } = keys(KeyConfig),
+     } = keys(),
     MaybeB58 = fun(undefined) -> undefined;
                   (Key) -> libp2p_crypto:pubkey_to_b58(Key)
                end,
-    MaybeUUIDv4 = fun(undefined) -> undefined;
-                  (Key) ->
-                      %% only production blackspots read onboarding_key from a file 
-                      %% and they're (unfortunately) considered to be uuids
-                      OnboardingFilename = get_onboarding_filename(),
-                      case filelib:is_file(OnboardingFilename) of
-                          true ->
-                              string:trim(binary_to_list(Key));
-                          false ->
-                              libp2p_crypto:pubkey_to_b58(Key)
-                      end
-               end,
     Props = [{pubkey, MaybeB58(PubKey)},
-             {onboarding_key, MaybeUUIDv4(OnboardingKey)}
```

Comment:
I don't think you can remove MaybeUUIDv4. There' s s set of "blackspots" out there that use UUIDs as their onboarding keys

Reply:
This code doesn't remove the ability for an onboarding key to be a UUIDv4. Instead, it shifts the responsibility of deciding whether it is a UUIDv4 or B58 to the place where it belongs: in the code that knows whether it is a UUIDv4 or a B58 key! Once the code reaches this point, `OnboardingKey` will already be in whatever format it needs to be in for dumping.

Reply:
And when I said:

>  ... in the code that knows whether it is a UUIDv4 or a B58 key!

The code that I was speaking of is `keys()/1`,

Reply:
Right, sorry, my point is that `keys/1` is somehow _not_ handling the fact that an onboarding key can be a uuid?

Reply:
Nope. I think the point will be best illustrated when we revisit what `MaybeUUIDv4` _was_ before I deleted it. It was badly named. This is what it did:

```
MaybeUUIDv4 = fun(undefined) -> undefined;		
                   (Key) ->		
                       %% only production blackspots read onboarding_key from a file       		
                       %% and they're (unfortunately) considered to be uuids		
                       OnboardingFilename = get_onboarding_filename(),		
                       case filelib:is_file(OnboardingFilename) of		
                           true ->		
                               string:trim(binary_to_list(Key));		
                           false ->		
                               libp2p_crypto:pubkey_to_b58(Key)		
                       end		
                end,
```

This has nothing to do with supporting UUIDv4 strings and everything to do with duplicating the logic of finding out where the onboarding key came from and _then_ turning it into a string.

Reply:
It's been a while, but I think I follow.. This code may not have been doing anything useful in practice, I guess. I just want to make sure that it doesn't fall over because the onboarding key is s a uuidv4 and keys/1 now does a pubkey_to_b58 by default. 

Perhaps get_onboarding_key/1 is the place where uuid onboarding keys are handled

Reply:
We just crossed comment paths. I updated my comment while you were commenting. The additions might make it more clear.

Reply:
> I just want to make sure that it doesn't fall over because the onboarding key is s a uuidv4 and keys/1 now does a pubkey_to_b58 by default.

I tested this code on a rak v1, which if you recall, really is a `blackspot`. The onboarding key there is a UUIDv4, which comes from the filesystem. `keys({file, ...})`. This causes `keys({file, ...})` to be run, which "does the right thing".

Reply:
Resolving as I think we're in agreement.